src/templates/data/global/content-types/builtin_carousel.js

```diff
@@ -2,6 +2,30 @@ const base = require('./_base.js')
 const Card = require('./builtin_card')
 const url = require('url')
 
+function render(data) {
```

Comment:
Shouldn't the choice for the kind of events to be returned (typing, text) be chosen by the api's caller? If someone creates a messenger channel on top of the api, he won't get the 'typing event', even if it may be wanted